src/main/java/net/dv8tion/jda/api/entities/Message.java

```diff
@@ -1056,6 +1057,78 @@ default MessageAction editMessage(@Nonnull MessageEmbed newContent)
+     *         the {@link net.dv8tion.jda.api.entities.TextChannel TextChannel}.</li>
+     *
+     *     <li>{@link net.dv8tion.jda.api.requests.ErrorResponse#UNKNOWN_MESSAGE UNKNOWN_MESSAGE}
+     *     <br>The edit was attempted after the Message had been deleted.</li>
+     * </ul>
+     *
+     * @param  actionRows
+     *         the new ActionRows (up to 5)
+     *
+     * @throws java.lang.UnsupportedOperationException
+     *         If this is a system message
+     * @throws java.lang.IllegalStateException
+     *         If the message attempting to be edited was not created by the currently logged in account, or
+     *         if any of the passed-in ActionRows is {@code null}
+     *
+     * @return {@link MessageAction MessageAction}
+     *         <br>The {@link net.dv8tion.jda.api.entities.Message Message} with the updated ActionRows
+     */
+    @Nonnull
+    @CheckReturnValue
+    MessageAction editComponents(@Nonnull Collection<? extends ComponentLayout> actionRows);
+
+    /**
+     * Edits this Message's ActionRows to the provided {@link net.dv8tion.jda.api.interactions.components.ActionRow ActionRows}.
+     * <br><b>Messages can only be edited by the account that sent them!</b>.
+     *
+     * <p>This message instance will not be updated by this operation, please use the response message instead.
+     *
+     * <p>The following {@link net.dv8tion.jda.api.requests.ErrorResponse ErrorResponses} are possible:
+     * <ul>
+     *     <li>{@link net.dv8tion.jda.api.requests.ErrorResponse#MISSING_ACCESS MISSING_ACCESS}
+     *     <br>The edit was attempted after the account lost access to the {@link net.dv8tion.jda.api.entities.Guild Guild}
+     *         typically due to being kicked or removed.</li>
+     *
+     *     <li>{@link net.dv8tion.jda.api.requests.ErrorResponse#MISSING_PERMISSIONS MISSING_PERMISSIONS}
+     *     <br>The edit was attempted after the account lost {@link net.dv8tion.jda.api.Permission#MESSAGE_WRITE Permission.MESSAGE_WRITE} in
+     *         the {@link net.dv8tion.jda.api.entities.TextChannel TextChannel}.</li>
+     *
+     *     <li>{@link net.dv8tion.jda.api.requests.ErrorResponse#UNKNOWN_MESSAGE UNKNOWN_MESSAGE}
+     *     <br>The edit was attempted after the Message had been deleted.</li>
+     * </ul>
+     *
+     * @param  actionRows
+     *         the new ActionRows (up to 5)
+     *
+     * @throws java.lang.UnsupportedOperationException
+     *         If this is a system message
+     * @throws java.lang.IllegalStateException
+     *         If the message attempting to be edited was not created by the currently logged in account, or
+     *         if any of the passed-in ActionRows is {@code null}
+     *
+     * @return {@link MessageAction MessageAction}
+     *         <br>The {@link net.dv8tion.jda.api.entities.Message Message} with the updated ActionRows
+     */
+    @Nonnull
+    @CheckReturnValue
+    MessageAction editComponents(@Nonnull ActionRow... actionRows);
```

Comment:
This should be a default implementation similar to the one in WebhookClient:

https://github.com/DV8FromTheWorld/JDA/blob/382f9c8fa7d6e2ba485d06812f8c31d23eaa3d09/src/main/java/net/dv8tion/jda/api/entities/WebhookClient.java#L759-L763